torchtitan/components/float8.py

```diff
@@ -109,13 +110,26 @@ def convert_to_float8_training(self, model: nn.Module):
         convert_to_float8_training(
             model,
             config=self.config,
-            module_filter_fn=lambda mod, fqn: fqn != "output",
+            module_filter_fn=self._module_filter_fn,
         )
         logger.info(
             "Swapped to Float8Linear layers with enable_fsdp_float8_all_gather="
             f"{self.config.enable_fsdp_float8_all_gather}"
         )
 
+    def _module_filter_fn(self, mod, fqn):
+        assert isinstance(mod, nn.Linear)
```

Comment:
We want to just return if it isn't a linear right?

Reply:
Oh yeah it already should skip non-linears [here](https://github.com/pytorch/ao/blob/5802d2d2ca9b762cf61be1f8b6bf3c587cb1bc91/torchao/float8/float8_linear_utils.py#L41) so I was just asserting to act as an alert if that accidentally changes.

I think you're right though that returning False here makes more sense, so it'll just not be converted in that case. Updated it accordingly.